verifier/src/channel.rs

```diff
@@ -177,9 +188,34 @@ impl<E: FieldElement, H: ElementHasher<BaseField = E::BaseField>> VerifierChanne
         let queries = self.trace_queries.take().expect("already read");
 
         // make sure the states included in the proof correspond to the trace commitment
-        for (root, proof) in self.trace_roots.iter().zip(queries.query_proofs.iter()) {
-            MerkleTree::verify_batch(root, positions, proof)
-                .map_err(|_| VerifierError::TraceQueryDoesNotMatchCommitment)?;
+
+        let items: Vec<H::Digest> =
+            { queries.main_states.rows().map(|row| H::hash_elements(row)).collect() };
+        <V as VectorCommitment<H>>::verify_many(
+            self.trace_commitments[0],
+            positions,
+            &items,
+            &queries.query_proofs[0],
+        )
+        .map_err(|_| VerifierError::TraceQueryDoesNotMatchCommitment)?;
+
+        if queries.aux_states.is_some() {
+            let items: Vec<H::Digest> = {
+                queries
+                    .aux_states
+                    .clone()
+                    .unwrap()
+                    .rows()
+                    .map(|row| H::hash_elements(row))
+                    .collect()
+            };
+            <V as VectorCommitment<H>>::verify_many(
+                self.trace_commitments[1],
+                positions,
+                &items,
+                &queries.query_proofs[1],
+            )
+            .map_err(|_| VerifierError::TraceQueryDoesNotMatchCommitment)?;
```

Comment:
I think we can simplify this as follows:
```rust
if let Some(ref aux_states) = queries.aux_states {
    let items: Vec<H::Digest> =
        aux_states.rows().map(|row| H::hash_elements(row)).collect();

    <V as VectorCommitment<H>>::verify_many(
        self.trace_commitments[1],
        positions,
        &items,
        &queries.query_proofs[1],
    )
    .map_err(|_| VerifierError::TraceQueryDoesNotMatchCommitment)?;
}
```

Reply:
thank you, changed now